src/petals/server/server.py

```diff
@@ -157,8 +157,8 @@ def __init__(
         if attn_cache_size is None:
             # Hidden size is 14336 for the bigscience/bloom-petals model. For other models, scale accordingly
             attn_cache_size = 0.5 * gib * num_blocks * self.block_config.hidden_size / 14336
+        self.attn_cache_size, self.alloc_timeout = attn_cache_size, alloc_timeout
```

Comment:
Now, we will create `MemoryCache` only in `ModuleContainer`, so it's garbage-collected with all remaining tensors when the container is destroyed during rebalancing.